src/main/java/org/disq_bio/disq/impl/file/FileSystemWrapper.java

```diff
@@ -39,6 +40,49 @@
  */
 public interface FileSystemWrapper extends Serializable {
 
+  /** Represents a file in a directory listing. */
+  class FileStatus implements Comparable<FileStatus> {
+    private final String path;
+    private final long length;
+
+    public FileStatus(String path, long length) {
+      this.path = path;
+      this.length = length;
+    }
+
+    /** @return the file path */
+    public String getPath() {
+      return path;
+    }
+
+    /** @return the length of the file in bytes */
+    public long getLength() {
+      return length;
+    }
+
+    @Override
+    public int compareTo(FileStatus o) {
+      int cmp = path.compareTo(o.path);
+      if (cmp != 0) {
+        return cmp;
+      }
+      return Long.compare(length, o.length);
+    }
+
+    @Override
+    public boolean equals(Object o) {
+      if (this == o) return true;
+      if (o == null || getClass() != o.getClass()) return false;
+      FileStatus that = (FileStatus) o;
+      return length == that.length && path.equals(that.path);
+    }
+
+    @Override
+    public int hashCode() {
+      return Objects.hash(path, length);
```

Comment:
`FileStatus` is immutable, `hashCode` could be created in the constructor and cached

Reply:
Not sure this change is needed - it seems like a bit of a premature optimization.